Source/WebCore/page/LocalFrameViewLayoutContext.cpp

```diff
@@ -660,6 +660,29 @@ void LocalFrameViewLayoutContext::popLayoutState()
     }
 }
 
+void LocalFrameViewLayoutContext::markNeedingTransformUpdate(RenderLayerModelObject* parent, RenderLayerModelObject* descendant)
+{
+    if (m_descendantsNeedingTransformUpdate.contains(parent)) {
```

Comment:
`contains/set` can be done at the same time with `ensure()`.

Reply:
We can't do this since we need to check if the descendant is in the map (to prevent adding it again after this gets called when the containing blocks layout finishes).